### PR TITLE
Add support for sequential RPCs across tasks

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: lib/rpc/rpc.toit [hunk 1]
```diff
@@ -2,31 +2,33 @@
...
-    if arguments is RpcSerializable: arguments = arguments.serialize_for_rpc
-    return synchronizer_.send -1: | id pid |
-      process_send_ pid SYSTEM_RPC_REQUEST_ [ id, name, arguments ]
-
-  invoke pid/int name/int arguments/any -> any:
+  invoke pid/int name/int arguments/any --sequential/bool -> any:
     if arguments is RpcSerializable: arguments = arguments.serialize_for_rpc
-    return synchronizer_.send pid: | id pid |
+    send ::= : synchronizer_.send pid: | id pid |
       process_send_ pid SYSTEM_RPC_REQUEST_ [ id, name, arguments ]
```
Comment (lask) [inline lib/rpc/rpc.toit:30]: I prefer this formatting as I find it more clear what belongs to the inner block.
```suggestion
    send ::= : 
      synchronizer_.send pid: | id pid |
        process_send_ pid SYSTEM_RPC_REQUEST_ [ id, name, arguments ]
```

File: tests/rpc_test.toit [hunk 2]
```diff
@@ -216,6 +217,41 @@ test_blocking myself/int broker/RpcBroker tasks/int [test] -> none:
...
+test_sequential myself/int broker/RpcBroker -> none:
+  tasks ::= 10
+  name ::= 800
+  latches ::= {:}
+  concurrency := 0
+  broker.register_procedure name:: | args |
+    result := null
+    try:
+      concurrency++
+      expect.expect_equals 1 concurrency  // Should be sequential!
```
Comment (lask) [inline tests/rpc_test.toit:229]: wouldn't it be better with a `yield` here to allow the other tasks to jump in?

**Review by lask — APPROVED** [review]
lgtm

File: tests/rpc_test.toit (hunk 2 above)
Comment (kasperl) [reply tests/rpc_test.toit:229]: The last.get call blocks until explicitly told to continue from the outside. That was intended to be (even) more robust than yielding.

File: lib/rpc/rpc.toit (hunk 1 above)
Comment (kasperl) [reply lib/rpc/rpc.toit:30]: Done.

File: tests/rpc_test.toit (hunk 2 above)
Comment (lask) [reply tests/rpc_test.toit:229]: Perfect.

**Pull request merged**